### ISSUE TITLE
Problem with Jest

### ISSUE THREAD

**Issue by desoss** [bpierre/dnum#11]
I cannot make my unit tests work.

I get this error:
```
Jest encountered an unexpected token

Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.
```
If I remove the lines of code that exploit dnum the tests work perfectly.

This is my jest.config.js:
```
module.exports = {
    preset: 'ts-jest',
    testEnvironment: 'node',
    roots: ["<rootDir>/src"],
    testMatch: ['**/*.test.(ts|tsx)'],
};
```

**Comment by bpierre** [comment]
Could you please post a reproduction somewhere? Thanks :pray:

**Comment by caffeinum** [comment]
I encounter this issue in a private repo. It's a nextjs typescript app that doesn't use `ts-jest`

jest.config.js:

```
import type { Config } from "jest";
import nextJest from "next/jest.js";

const createJestConfig = nextJest({
  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
  dir: "./",
});

// Add any custom config to be passed to Jest
const config: Config = {
  coverageProvider: "v8",
  testEnvironment: "jsdom",
  // Add more setup options before each test is run
  setupFilesAfterEnv: ["<rootDir>/jest.setup.ts"],
  moduleNameMapper: {
    "^~/(.*)$": "<rootDir>/src/$1",
  },
  maxWorkers: "50%",
  workerThreads: true,
};

// createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async
export default createJestConfig(config);

```

**Comment by caffeinum** [comment]
Update: I was able to solve this issue by switching from `jest` to `vitest`. Even though initially it gave the same error, The latter has a workaround that fixed the issue for me:

```typescript
vi.mock("dnum", () => vi.importActual("dnum"));
```

I have added this line to the top of my test file and for some reason this imports the library correctly.

I didn't have many tests in my project, so re-writing them was basically finding the right config + adding `vitest` imports on top of each test file. This might not work for everyone.

Another solution we have found is to literally fork `dnum` and re-export it using `@tanstack/config`. See https://github.com/bpierre/dnum/pull/17, thanks @freeatnet 

Published to: https://www.npmjs.com/package/@sloikaxyz/dnum
